src/TurboMqtt/IO/Tcp/FakeMqttTcpServer.cs

```diff
@@ -157,6 +158,8 @@ private async Task ProcessClientAsync(Socket socket)
                 CancellationTokenSource.CreateLinkedTokenSource(clientShutdownCts.Token, _shutdownTcs.Token);
             while (!linkedCts.IsCancellationRequested)
             {
+                if (closed)
+                    break;
```

Comment:
Clean up fake server code, if we're already closed, just quit

Reply:
LGTM